src/components/ChatView.tsx

```diff
@@ -48,6 +48,8 @@ export default function ChatView (props: Props) {
     }
   })
 
+  const [shouldShowOldMessages, setShouldShowOldMessages] = React.useState(false)
```

Comment:
I see a lot of context/reducer usage in this app to store app state. This felt like local state that the rest of the app outside chat didn't need knowledge. Not sure if that's inline with the state management strategy here though!